dynawo/sources/Solvers/SolverTRAP/DYNSolverTRAP.cpp

```diff
@@ -0,0 +1,615 @@
+//
+// Copyright (c) 2015-2019, RTE (http://www.rte-france.com)
+// See AUTHORS.txt
+// All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, you can obtain one at http://mozilla.org/MPL/2.0/.
+// SPDX-License-Identifier: MPL-2.0
+//
+// This file is part of Dynawo, an hybrid C++/Modelica open source time domain
+// simulation tool for power systems.
+//
+
+/**
+ * @file  DYNSolverTRAP.cpp
+ *
+ * @brief Trapezoidal solver implementation
+ *
+ * Trapezoidal solver is based on Trapezoidal Method that is nearly the same as Back Euler one.
+ *
+ */
+
+#include "DYNSolverTRAP.h"
+
+#include <cmath>
+#include <iostream>
+#include <iomanip>
+#include <vector>
+
+#include <boost/shared_ptr.hpp>
+
+#include "PARParametersSet.h"
+#include "PARParameter.h"
+
+#include "DYNMacrosMessage.h"
+#include "DYNSolverKINEuler.h"
+#include "DYNSolverKINAlgRestoration.h"
+#include "DYNTrace.h"
+#include "DYNModel.h"
+
+using boost::shared_ptr;
+using std::endl;
+using std::max;
+using std::min;
+using std::ofstream;
+using std::stringstream;
+using std::set;
+using std::setw;
+using std::setfill;
+using std::vector;
+using std::map;
+using std::string;
+using std::make_pair;
+
+using parameters::ParametersSet;
+using timeline::Timeline;
+
+/**
+ * @brief SolverTRAPFactory getter
+ * @return A pointer to a new instance of SolverTRAPFactory
+ */
+extern "C" DYN::SolverFactory* getFactory() {
+  return (new DYN::SolverTRAPFactory());
+}
+
+/**
+ * @brief SolverTRAPFactory destroy method
+ */
+extern "C" void deleteFactory(DYN::SolverFactory* factory) {
+  delete factory;
+}
+
+/**
+ * @brief SolverTRAP getter
+ * @return A pointer to a new instance of SolverTRAP
+ */
+extern "C" DYN::Solver* DYN::SolverTRAPFactory::create() const {
+  DYN::Solver* solver(new DYN::SolverTRAP());
+  return solver;
+}
+
+/**
+ * @brief SolverTRAP destroy method
+ */
+extern "C" void DYN::SolverTRAPFactory::destroy(DYN::Solver* solver) const {
+  delete solver;
+}
+
+namespace DYN {
+
+SolverTRAPFactory::SolverTRAPFactory() {
```

Comment:
default constructor and destructors: no need to declare them and use the default one